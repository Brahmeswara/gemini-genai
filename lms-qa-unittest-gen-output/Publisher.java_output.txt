```java
package com.example.lws.models;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PublisherTest {

    private Publisher publisher;

    @BeforeEach
    void setUp() {
        publisher = Publisher.builder()
                .id(1L)
                .name("John Doe")
                .address("123 Main Street")
                .phone("555-555-5555")
                .email("john.doe@example.com")
                .build();
    }

    @Test
    void getId() {
        assertEquals(1L, publisher.getId());
    }

    @Test
    void getName() {
        assertEquals("John Doe", publisher.getName());
    }

    @Test
    void getAddress() {
        assertEquals("123 Main Street", publisher.getAddress());
    }

    @Test
    void getPhone() {
        assertEquals("555-555-5555", publisher.getPhone());
    }

    @Test
    void getEmail() {
        assertEquals("john.doe@example.com", publisher.getEmail());
    }

    @Test
    void testToString() {
        String expected = "Publisher(id=1, name=John Doe, address=123 Main Street, phone=555-555-5555, email=john.doe@example.com)";
        assertEquals(expected, publisher.toString());
    }
}
```