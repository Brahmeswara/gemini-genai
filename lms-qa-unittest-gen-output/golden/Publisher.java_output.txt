```java
package com.example.lws.models;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class PublisherUnitTest {

    @Test
    void testBuilder() {
        Publisher publisher = Publisher.builder()
                .name("Test Publisher")
                .address("123 Main Street")
                .phone("555-1212")
                .email("test@publisher.com")
                .build();

        assertEquals("Test Publisher", publisher.getName());
        assertEquals("123 Main Street", publisher.getAddress());
        assertEquals("555-1212", publisher.getPhone());
        assertEquals("test@publisher.com", publisher.getEmail());
    }

    @Test
    void testGetterAndSetter() {
        Publisher publisher = new Publisher();
        publisher.setName("Test Publisher");
        publisher.setAddress("123 Main Street");
        publisher.setPhone("555-1212");
        publisher.setEmail("test@publisher.com");

        assertEquals("Test Publisher", publisher.getName());
        assertEquals("123 Main Street", publisher.getAddress());
        assertEquals("555-1212", publisher.getPhone());
        assertEquals("test@publisher.com", publisher.getEmail());
    }
}
```